src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -2257,14 +2256,13 @@ def check_cl_registration_and_get_oid(cmd, cl_oid, subscription_id):
 
 def get_custom_locations_oid(cmd, cl_oid):
     try:
-        sp_graph_client = get_graph_client_service_principals(cmd.cli_ctx)
-        sub_filters = []
-        sub_filters.append("appId eq '{}'".format("bc313c14-388c-4e7d-a58e-70017303ee3b"))  # Lookup using well-known first-party application id
-        result = list(sp_graph_client.list(filter=(' and '.join(sub_filters))))
-        if len(result) != 0:
-            if cl_oid is not None and cl_oid != result[0].object_id:
-                logger.debug("The 'Custom-locations' OID passed is different from the actual OID({}) of the Custom Locations RP app. Proceeding with the correct one...".format(result[0].object_id))
-            return result[0].object_id  # Using the fetched OID
+        graph_client = graph_client_factory(cmd.cli_ctx)
+        app_id = "bc313c14-388c-4e7d-a58e-70017303ee3b"
+        app_object = graph_client.service_principal_list(filter="appId eq '{}'".format(app_id))
```

Comment:
Should we handle the case where caller does not have `Application.Read.All` permission on graph ?

IMO this should be wrap into a `try` `catch` block and returns an actionable error message.

Reply:
just checked the warning message the next line, we might just need to update the doc and add a prerequisites about this.

Remind me the current flow, if we do return `""` is the whole operation is failing ?

Reply:
Custom Locations is enabled by default. If user doesn't have permission for retrieving the service principal via MS Graph, then it won't be enabled, and a warning message will be printed, which is the current behavior.

Reply:
Make senses. Is just that would be great to provide more insights about the real failure here, if this is a graph permission issue or just that the call did returns no `oid`.
Can we ask PM to edit the doc : https://aka.ms/CustomLocationsObjectID to makes sure we ask customer to make sure they are having the `Application.Read.All` permission.

Reply:
Actually I just checked and looks like `Application.Read.All` is granted by default on any default AAD User

Reply:
For service principals, they need to be granted the permission explicitly by an admin.